### PR TITLE
Use `dbt.string_literal` behind the scenes for quoting non-null values

### PR DESCRIPTION
**Pull request by b-per**
Alternative to #419 for wrapping strings with quotes

Add test for #421 which is failing for BQ without the fix

### CODE REVIEW

**Comment by b-per** [comment]
I think that our test Redshift server is down and the latest databricks version is still not working with pandas >= 2.2

But the test passes on BigQuery! 😄

File: integration_tests/models/staging/source_1/schema.yml [hunk 1]
```diff
@@ -5,7 +5,9 @@ models:
     description: test
     columns:
       - name: id 
-        description: hocus pocus
+        description: >
```
Comment (graciegoheen) [inline integration_tests/models/staging/source_1/schema.yml:8]: lol

File: integration_tests/models/dbt_project_evaluator_schema_tests/core.yml [hunk 2]
```diff
@@ -17,10 +17,6 @@ models:
         tests:
           - unique
           - not_null
-      - name: on_schema_change
```
Comment (graciegoheen) [inline integration_tests/models/dbt_project_evaluator_schema_tests/core.yml:20]: why remove this test?

File: macros/wrap_string_with_quotes.sql [hunk 3]
```diff
@@ -1,7 +1,14 @@
...
+  {% if str is none %}
     {{ return('cast(NULL as ' ~ dbt.type_string() ~ ')') }}
   {% else %}
-    {{ return("'" ~ str ~ "'") }}
+    {{ dbt.string_literal(str) }}
   {% endif %}
 {% endmacro %}
+
+{#
+  To be removed when https://github.com/dbt-labs/dbt-bigquery/pull/1089 is merged
```
Comment (graciegoheen) [inline macros/wrap_string_with_quotes.sql:10]: Let's touch base with adapters team to see if this is something we can merge asap

File: models/marts/dag/fct_hard_coded_references.sql [hunk 4]
```diff
@@ -11,7 +11,7 @@ final as (
         resource_name as model,
         hard_coded_references
     from models
-    where hard_coded_references is not null
+    where hard_coded_references != ''
```
Comment (graciegoheen) [inline models/marts/dag/fct_hard_coded_references.sql:14]: why the change from null to empty string?

Comment (b-per) [reply models/marts/dag/fct_hard_coded_references.sql:14]: Because I changed `wrap_string_with_quotes` to handle `none` and `''` separately

File: integration_tests/models/dbt_project_evaluator_schema_tests/core.yml (hunk 2 above)
Comment (b-per) [reply integration_tests/models/dbt_project_evaluator_schema_tests/core.yml:20]: Because it failed 🤣 

If I recall, this is because we now handle empty strings or None differently in `wrap_string_with_quotes(str)`

**Comment by b-per** [comment]
🤔 Why is this failing on `fct_test_directories` now...

**Comment by b-per** [comment]
OK, got it. 

A side effect of null vs empty string handling.

**Review by dave-connors-3 — APPROVED** [review]
